### PR TITLE
Talkie for Auto, Stun, Kill

### PR DESCRIPTION
**Pull request by NoSloppy**
Addressing // TODO: would rather do a Talkie to speak the mode we're in after mode sound

### CODE REVIEW

File: props/blaster.h [hunk 1]
```diff
@@ -306,8 +306,19 @@ class Blaster : public PropBase {
 	  hybrid_font.PlayCommon(&SFX_mode);
 	  return;
 	}
-	// TODO: would rather do a Talkie to speak the mode we're in after mode sound
-	beeper.Beep(0.05, 2000.0);
+	if (blaster_mode == MODE_STUN) {
+	  talkie.Say(spSTUN); return;
```
Comment (profezzorn) [inline props/blaster.h:310]: Fonts should have the opportunity to override these effects.

File: props/blaster.h [hunk 2]
```diff
@@ -306,8 +306,19 @@ class Blaster : public PropBase {
 	  hybrid_font.PlayCommon(&SFX_mode);
 	  return;
 	}
-	// TODO: would rather do a Talkie to speak the mode we're in after mode sound
-	beeper.Beep(0.05, 2000.0);
+	if (blaster_mode == MODE_STUN) {
```
Comment (profezzorn) [inline props/blaster.h:309]: use a switch()

File: props/blaster.h (hunk 1 above)
Comment (NoSloppy) [reply props/blaster.h:310]: Are you suggesting we should check for a font version of the spoken words? 
Since there already is a stun.wav that is the blaster sound, something such as "spstun" or similar?

Comment (profezzorn) [reply props/blaster.h:310]: How about mdstun, mdauto, mdkill ?

Comment (NoSloppy) [reply props/blaster.h]: Changed to mdstun, mdauto, mdkill. Removed plain "mode" as these replace that with specific sounds for each mode, yes?

File: props/blaster.h (hunk 2 above)
Comment (NoSloppy) [reply props/blaster.h]: Done.

File: props/blaster.h [hunk 3]
```diff
@@ -301,22 +303,32 @@ class Blaster : public PropBase {
...
       case EFFECT_PLI_OFF: hybrid_font.PlayCommon(&SFX_plioff); return;
-	
+  
+    }
+  }
+
+  void SayMode() {
+    switch(blaster_mode) {
+      case MODE_STUN:
+      if (SFX_mdstun) hybrid_font.PlayCommon(&SFX_mdstun); return;
```
Comment (profezzorn) [inline props/blaster.h:321]: Did you test it?
(Hint; thjis doesn't work.)

Comment (NoSloppy) [reply props/blaster.h]: I was dumb and lazy and just commented out the font files checks to avoid a song and dance to delete the files.
That being said, it's corrected and tested properly now.
Why wouldn't it fall through to the talkie if no sounds were found? I was just trying to condense the if clause to one line for neatness/efficiency.

File: props/blaster.h [hunk 4]
```diff
@@ -301,22 +307,44 @@ class Blaster : public PropBase {
...
       case EFFECT_UNJAM: hybrid_font.PlayCommon(&SFX_unjam); return;
       case EFFECT_PLI_ON: hybrid_font.PlayCommon(&SFX_plion); return;
       case EFFECT_PLI_OFF: hybrid_font.PlayCommon(&SFX_plioff); return;
-	
+  
+    }
+  }
+
+  void SayMode() {
+    switch(blaster_mode) {
```
Comment (profezzorn) [inline props/blaster.h:323]: space after switch

File: props/blaster.h [hunk 5]
```diff
@@ -301,22 +307,44 @@ class Blaster : public PropBase {
...
+      case MODE_STUN:
+        if (SFX_mdstun) {
+          hybrid_font.PlayCommon(&SFX_mdstun);
+          return;
```
Comment (profezzorn) [inline props/blaster.h:327]: This return is redundant.

File: props/blaster.h [hunk 6]
```diff
@@ -301,22 +307,44 @@ class Blaster : public PropBase {
...
+        } else {
+          talkie.Say(spSTUN);
+        }
+      return;
```
Comment (profezzorn) [inline props/blaster.h:331]: this should be "break", not "return"

File: props/blaster.h [hunk 7]
```diff
@@ -301,22 +307,44 @@ class Blaster : public PropBase {
...
+      case MODE_KILL:
+        if (SFX_mdkill) {
+          hybrid_font.PlayCommon(&SFX_mdkill);
+          return;
```
Comment (profezzorn) [inline props/blaster.h:335]: redundant

File: props/blaster.h [hunk 8]
```diff
@@ -301,22 +307,44 @@ class Blaster : public PropBase {
...
+        } else {
+          talkie.Say(spKILL);      
+        }
+      return;
```
Comment (profezzorn) [inline props/blaster.h:339]: break

File: props/blaster.h [hunk 9]
```diff
@@ -301,22 +307,44 @@ class Blaster : public PropBase {
...
+      case MODE_AUTO:
+        if (SFX_mdauto) {
+          hybrid_font.PlayCommon(&SFX_mdauto);
+          return;
```
Comment (profezzorn) [inline props/blaster.h:343]: redundant

File: props/blaster.h [hunk 10]
```diff
@@ -301,22 +307,44 @@ class Blaster : public PropBase {
...
+        } else {
+          talkie.Say(spAUTOFIRE);       
+        }
+      return;
```
Comment (profezzorn) [inline props/blaster.h:347]: break

File: props/blaster.h [hunk 11]
```diff
@@ -10,13 +10,15 @@ EFFECT(clipout);
 EFFECT(empty);
 EFFECT(full);
 EFFECT(jam);
-EFFECT(mode);
```
Comment (profezzorn) [inline props/blaster.h:13]: I wonder if the "mode" sound would be preferable to the talkie effects?
At the very least, maybe should add a comment at the beginning of the file telling people that they can copy the mode sound to mdstun/mdkill/mdauto if they prefer one sound over the talkie effects.

Comment (NoSloppy) [reply props/blaster.h:13]: How about an else if to just use "mode" for all 3 as backup
check for mdkill
if not check for mode
if not play talkie version?

Comment (profezzorn) [reply props/blaster.h:13]: that would also work, and may be preferable.

File: props/blaster.h [hunk 12]
```diff
@@ -30,7 +35,7 @@ class Blaster : public PropBase {
     MODE_AUTO
   };
 
-  BlasterMode blaster_mode = MODE_STUN;
+  BlasterMode blaster_mode = MODE_KILL;
```
Comment (profezzorn) [inline props/blaster.h:38]: This was not part of the PR description.

File: props/blaster.h (hunk 4 above)
Comment (NoSloppy) [reply props/blaster.h]: This whole section fixed and tested good.

File: props/blaster.h (hunk 12 above)
Comment (NoSloppy) [reply props/blaster.h:38]: Oops. Sorry. Ok do have it default start on kill?

Comment (profezzorn) [reply props/blaster.h:38]: Let's not change it unless there is a reason to change it.
